commit ed5c3a29cbff4fd4cf4c239b62ccb8c47a96cab6
Author: asaveljevs <asaveljevs@97f52cf1-0a1b-0410-bd0e-c28be96e8082>
Date:   Tue Aug 31 09:17:26 2010 +0000

     - [ZBX-2945] email sender now knows how to handle multiline responses from SMTP servers (asaveljevs)
    	Reintegrated dev/zbx-2945-multiline-response into 1.8.
    
    
    git-svn-id: svn://svn.zabbix.com/branches/1.8@14211 97f52cf1-0a1b-0410-bd0e-c28be96e8082

diff --git a/src/libs/zbxemail/email.c b/src/libs/zbxemail/email.c
index 57ba4ec..4193635 100644
--- a/src/libs/zbxemail/email.c
+++ b/src/libs/zbxemail/email.c
@@ -47,21 +47,28 @@
  */
 ssize_t smtp_readln(int fd, char *buf, int buf_len)
 {
-	ssize_t	nbytes, read_bytes = 0;
+	ssize_t	nbytes, read_bytes;
+
+	buf_len--;	/* '\0' */
 
-	buf_len --;	/* '\0' */
 	do
 	{
-		if (-1 == (nbytes = read(fd, &((char*)buf)[read_bytes], 1)))
-			return nbytes;				/* Error */
+		read_bytes = 0;
 
-		read_bytes += nbytes;
-	}
-	while (nbytes > 0 &&					/* End Of File (socket closed) */
-			read_bytes < buf_len &&			/* End of Buffer */
-			((char*)buf)[read_bytes - 1] != '\n' );	/* new line */
+		do
+		{
+			if (-1 == (nbytes = read(fd, &((char*)buf)[read_bytes], 1)))
+				return nbytes;				/* Error */
 
-	buf[read_bytes] = '\0';
+			read_bytes += nbytes;
+		}
+		while (nbytes > 0 &&					/* End Of File (socket closed) */
+				read_bytes < buf_len &&			/* End of Buffer */
+				((char*)buf)[read_bytes - 1] != '\n' );	/* new line */
+
+		buf[read_bytes] = '\0';
+	}
+	while (read_bytes >= 4 && isdigit(buf[0]) && isdigit(buf[1]) && isdigit(buf[2]) && buf[3] == '-');
 
 	return read_bytes;
 }
